store/src/transaction.rs

```diff
@@ -79,135 +79,138 @@ impl StoreTransaction {
     pub fn get_update_for_tip_hash(
         &self,
         snapshot: &RocksDBTransactionSnapshot<'_>,
-    ) -> Option<H256> {
+    ) -> Option<packed::Byte32> {
         self.inner
             .get_for_update(COLUMN_META, META_TIP_HEADER_KEY, snapshot)
             .expect("db operation should be ok")
-            .map(|slice| H256::from_slice(&slice.as_ref()[..]).expect("db safe access"))
+            .map(|slice| {
+                packed::Byte32Reader::from_slice(&slice.as_ref()[..])
+                    .should_be_ok()
+                    .to_entity()
+            })
     }
 
-    pub fn insert_tip_header(&self, h: &Header) -> Result<(), Error> {
-        self.insert_raw(COLUMN_META, META_TIP_HEADER_KEY, h.hash().as_bytes())
+    pub fn insert_tip_header(&self, h: &HeaderView) -> Result<(), Error> {
+        self.insert_raw(COLUMN_META, META_TIP_HEADER_KEY, h.hash().as_slice())
     }
 
-    pub fn insert_block(&self, block: &Block) -> Result<(), Error> {
-        let hash = block.header().hash().as_bytes();
-        {
-            let builder = protos::StoredHeader::full_build(block.header());
-            self.insert_raw(COLUMN_BLOCK_HEADER, hash, builder.as_slice())?;
-        }
-        {
-            let builder = protos::StoredUncleBlocks::full_build(block.uncles());
-            self.insert_raw(COLUMN_BLOCK_UNCLE, hash, builder.as_slice())?;
-        }
-        {
-            let builder = protos::StoredProposalShortIds::full_build(block.proposals());
-            self.insert_raw(COLUMN_BLOCK_PROPOSAL_IDS, hash, builder.as_slice())?;
-        }
-        // key len: 32 (block_hash) + 4 (index)
-        let mut store_key = Vec::with_capacity(36);
-        store_key.extend_from_slice(hash);
-        for (index, tx) in block.transactions().iter().enumerate() {
-            let builder = protos::StoredTransaction::full_build(tx);
-            store_key.splice(32.., (index as u32).to_be_bytes().iter().cloned());
-            self.insert_raw(COLUMN_BLOCK_BODY, &store_key, builder.as_slice())?;
+    pub fn insert_block(&self, block: &BlockView) -> Result<(), Error> {
+        let hash = block.hash();
+        let header = block.header().pack();
+        let uncles = block.uncles().pack();
+        let proposals = block.data().proposals();
+        self.insert_raw(COLUMN_BLOCK_HEADER, hash.as_slice(), header.as_slice())?;
+        self.insert_raw(COLUMN_BLOCK_UNCLE, hash.as_slice(), uncles.as_slice())?;
+        self.insert_raw(
+            COLUMN_BLOCK_PROPOSAL_IDS,
+            hash.as_slice(),
+            proposals.as_slice(),
+        )?;
+        for (index, tx) in block.transactions().into_iter().enumerate() {
+            let key = packed::TransactionKey::new_builder()
+                .block_hash(hash.clone())
+                .index(index.pack())
```

Comment:
This `index` was defined as `BeUint32` in schema file, and I implement conversions between `u32.to_be_bytes()` and `BeUint32`. And the `TransactionKey` was defined as `struct` in schema file, so the `key.as_slice()` can use for prefix-search.